RedacteurPortaal.Api/Controllers/NewsItemController.cs

```diff
@@ -26,99 +27,78 @@ public NewsItemController(ILogger<NewsItemController> logger, IGrainManagementSe
     }
 
     [HttpPost]
-    public async Task<IActionResult> SaveNewsItem([FromBody] NewsItemDetailDto newsitem)
+    public async Task<ActionResult<NewsItemDetailDto>> SaveNewsItem([FromBody] NewsItemDetailDto newsitem)
     {
-        var newguid = Guid.NewGuid();
+        Guid newguid = Guid.NewGuid();
+
         TypeAdapterConfig<NewsItemDetailDto, NewsItemModel>
             .NewConfig()
-            .Map(dest => dest.ContactDetails,
-                src => src.ContactDetails.AsQueryable().ProjectToType<Contact>(null).ToList())
-            .Map(dest => dest.LocationDetails,
-                src => src.LocationDetails.Adapt<Location>())
-            .Map(dest => dest.Body,
-                src => src.Body != null ? src.Body.Adapt<ItemBody>() : new ItemBody())
-            .Map(dest => dest.Source,
-                src => src.Source != null ? src.Source.Adapt<FeedSource>() : new FeedSource())
-            .Map(dest => dest.Videos,
-                src => src.Videos.AsQueryable().ProjectToType<MediaVideoItem>(null).ToList())
-            .Map(dest => dest.Audio,
-                src => src.Audio.AsQueryable().ProjectToType<MediaAudioItem>(null).ToList())
-            .Map(dest => dest.Photos,
-                src => src.Photos.AsQueryable().ProjectToType<MediaPhotoItem>(null).ToList())
             .Map(dest => dest.Id,
                 src => newguid);
-        
+
         TypeAdapterConfig<MediaVideoItemDto, MediaVideoItem>
             .NewConfig()
             .Map(dest => dest.Duration,
                   src => TimeSpan.FromSeconds(src.DurationSeconds));
 
         const string successMessage = "News item was created";
-        var grain = await this.grainService.GetGrain(newguid);
-
+        var grain = await this.grainService.CreateGrain(newguid);
         var update = newsitem.Adapt<NewsItemModel>();
         await grain.Update(update);
+
+        var createdGrain = await this.grainService.GetGrain(newguid);
+        var createdItem = await createdGrain.Get();
```

Comment:
Not necessary

Reply:
Hier ook dat de grain update functie de correcte versie van de state terugstuurt